src/bun.js/webcore/streams.zig

```diff
@@ -3761,6 +3761,10 @@ pub const ReadResult = union(enum) {
                     StreamResult{ .owned = bun.ByteList.init(slice) }
                 else if (done)
                     StreamResult{ .into_array_and_done = .{ .len = @as(Blob.SizeType, @truncate(slice.len)), .value = view } }
+                else if (view == .zero and done)
```

Comment:
probably not an issue, but you have this == .zero check twice, it could be a nested if?